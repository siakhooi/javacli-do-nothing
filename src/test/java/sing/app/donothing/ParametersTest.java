package sing.app.donothing;

import static org.junit.jupiter.api.Assertions.assertEquals;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.PrintStream;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;

@ExtendWith({SnapshotExtension.class})
public class ParametersTest {

  PrintStream stdout;
  ByteArrayOutputStream baos;
  private Expect expect;

  @BeforeEach
  void setup() {
    stdout = System.out;
    baos = new ByteArrayOutputStream();
    System.setOut(new PrintStream(baos));
  }

  @AfterEach
  void teardown() throws IOException {
    System.setOut(stdout);
    baos.close();
  }
  
  @Test
  void testPrintUsage() {
    Parameters p=new Parameters();
    p.process(new String[]{"-h"});
    p.printUsage();
    expect.toMatchSnapshot(baos.toString());
  }

  @Test
  void testProcess1() {
    Parameters p=new Parameters();
    p.process(new String[]{"-h"});
    assertEquals(true, p.help);
    assertEquals(false, p.version);
    assertEquals(0, p.parameters.size());
  }
  @Test
  void testProcess2() {
    Parameters p=new Parameters();
    p.process(new String[]{"-v"});
    assertEquals(false, p.help);
    assertEquals(true, p.version);
    assertEquals(0, p.parameters.size());
  }
  @Test
  void testProcess3() {
    Parameters p=new Parameters();
    p.process(new String[]{"abc", "cde"});
    assertEquals(false, p.help);
    assertEquals(false, p.version);
    assertEquals(2, p.parameters.size());
  }
}
