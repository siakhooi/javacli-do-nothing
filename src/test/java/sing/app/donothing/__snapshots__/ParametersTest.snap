sing.app.donothing.ParametersTest.testPrintUsage=[
Usage: java-do-nothing [options]
  Options:
    --version, -v
      Display version
      Default: false
    --help, -h
      Display help
.
.
]